File: Robi - Digi-thor - Assessment/C - Forward Backward Moving.cpp
# include<bits/stdc++.h>

using namespace std;

int main(){
    int arrSz, rotateN, rotateT;

    cin >> arrSz >> rotateN >> rotateT;

    vector<int> vec;

    for(int i=0; i< arrSz; i++){
        int elem;
        cin >> elem;
        vec.push_back(elem);
    }

    if(rotateT == 0){
        rotate(vec.begin(), vec.begin()+rotateN, vec.end());
    }
    else{
        for(int i=0; i<rotateN; i++)
            rotate(vec.begin(), vec.end()-1, vec.end());
    }

    for(int i=0; i< arrSz-1; i++)
        cout << vec[i] << " ";
    cout << vec[arrSz-1];

    return 0;
}

File: Robi - Digi-thor - Assessment/C - Forward Backward Moving.py
arrSz, rotateN, rotateT = tuple(map(int, input().split()))

arr = list(map(int, input().split()))

if rotateT == 0:
    # left rotate
    arr = [arr[(i + rotateN) % len(arr)] for i, x in enumerate(arr)]
else:
    # right rotate
    arr = [arr[(i - rotateN) % len(arr)] for i, x in enumerate(arr)]

print(" ".join(map(str, arr)))